receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -64,6 +67,56 @@ func (prw *prometheusRemoteWriteReceiver) Shutdown(ctx context.Context) error {
 	return prw.server.Shutdown(ctx)
 }
 
-func (prw *prometheusRemoteWriteReceiver) handlePRW(_ http.ResponseWriter, _ *http.Request) {
-	fmt.Println("handleWrite called")
+func (prw *prometheusRemoteWriteReceiver) handlePRW(w http.ResponseWriter, req *http.Request) {
+	contentType := req.Header.Get("Content-Type")
+	if contentType == "" {
+		prw.settings.Logger.Warn("message received without Content-Type header, rejecting")
+		http.Error(w, "Content-Type header is required", http.StatusUnsupportedMediaType)
+		return
+	}
+
+	msgType, err := prw.parseProto(contentType)
+	if err != nil {
+		prw.settings.Logger.Warn("Error decoding remote-write request", zapcore.Field{Key: "error", Type: zapcore.ErrorType, Interface: err})
+		http.Error(w, err.Error(), http.StatusUnsupportedMediaType)
+		return
+	}
+	if msgType != promConfig.RemoteWriteProtoMsgV2 {
+		prw.settings.Logger.Warn("message received with unsupported proto version, rejecting")
+		http.Error(w, "Unsupported proto version", http.StatusUnsupportedMediaType)
+		return
+	}
+
+	// After parsing the content-type header, the next step would be to handle content-encoding.
```

Comment:
nit: can you change this to a list of TODOs so that it is clear what still needs to be done?

Reply:
This is not a TODO. OTel's HTTP implementation has a middleware that already decompresses and I was happily surprised by that. This comment was just to tell people something like "The RW spec says requests come in compressed format, but we don't need to handle that"

Reply:
Makes sense :)